src/main/java/duke/storage/DukeDecoder.java

```diff
@@ -0,0 +1,78 @@
+package src.main.java.duke.storage;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import src.main.java.duke.data.Duke;
+import src.main.java.duke.data.exception.IllegalValueException;
+import src.main.java.duke.data.task.Deadline;
+import src.main.java.duke.data.task.Event;
+import src.main.java.duke.data.task.Task;
+import src.main.java.duke.data.task.TaskList;
+import src.main.java.duke.data.task.Todo;
+import src.main.java.duke.storage.StorageFile.StorageOperationException;
+
+
+/**
+ * DukeDencoder decodes all the {@code Task} in the {@code toSave} into a list of decodable
```

Comment:
Nice use of pointers to other parts of the code! I did not know this existed. Thanks for showing this